apps/prairielearn/src/sync/course-db.ts

```diff
@@ -1136,13 +1148,24 @@ function validateAssessment({
             maxAutoPoints: zoneQuestion.maxAutoPoints,
             autoPoints: zoneQuestion.autoPoints,
             manualPoints: zoneQuestion.manualPoints,
+            allowRealTimeGrading: effectiveAlternativeGroupAllowRealTimeGrading,
           },
         ];
       } else {
         errors.push('Zone question must specify either "alternatives" or "id"');
       }
 
       alternatives.forEach((alternative) => {
+        if (
+          !alternative.allowRealTimeGrading &&
+          ((Array.isArray(alternative.autoPoints) && alternative.autoPoints.length > 1) ||
+            (Array.isArray(alternative.points) && alternative.points.length > 1))
+        ) {
+          errors.push(
+            'Cannot specify an array of multiple point values if real-time grading is disabled',
```

Comment:
The original error messages differentiated between question points and alternative points; this new one does not. This is partly because it was easier to implement and partly because I think the distinction doesn't really matter to users, especially now that real-time grading can be disabled at any level of the hierarchy.

This will also show the error message once per "assessment question" that violates this rule. This may not be super useful because these errors don't have any position information associated with them, so it'd just repeat the same text over and over. It'd be completely possible to record the error at most once per assessment.

Reply:
It may be helpful to include the full path of the violation in the error itself? That way it is easier to fix. Bonus points if you can do that for all `errors`, (e.g. a `path` associated with an error)

Reply:
That sounds good.

Reply:
I opened https://github.com/PrairieLearn/PrairieLearn/issues/12846 to track this.